### DISCUSSION TITLE
Where could I download yolo11m.yaml

### DISCUSSION THREAD

**Discussion by Lingpheng** [ultralytics/ultralytics#20847] [Q&A] [answered]
I dont want to pretrain my model, so i want to load a new model like this : 

model = YOLO("ultralytics/cfg/models/11/yolo11.yaml")
results = model.train(data=datas.yaml, epochs=50, patience=15,)
                      
but in this way, i didnt find yolo11m or n or s, do they exist or i dont need to specify it.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Lingpheng, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response—an Ultralytics engineer will review your question and assist you soon!

**Reply by glenn-jocher** [reply, marked as answer]
Hi @Lingpheng! The YOLO11 model variants (n, s, m, l, x) are defined by their depth and width multipliers within the main `yolo11.yaml` configuration file. You don't need separate YAML files for each variant.

To train from scratch without pretrained weights, you can use:

```python
from ultralytics import YOLO

# Load architecture only (no pretrained weights)
model = YOLO("yolo11n.yaml")  # or yolo11s.yaml, yolo11m.yaml, etc.
results = model.train(data="datas.yaml", epochs=50, patience=15)
```

The model variant YAML files (yolo11n.yaml, yolo11m.yaml, etc.) are located in the `ultralytics/cfg/models/11/` directory and define the specific architecture parameters for each model size.

**Reply by Lingpheng** [reply]
Hi @glenn-jocher , Thanks for your fast reply. I tried the yolo11m.yaml and yolo11n.yaml as you said, it does work!!!

**Reply by glenn-jocher** [reply]
You're welcome @Lingpheng! Glad I could help you get it working. The different YAML files make it easy to choose the model size that best fits your computational resources and accuracy requirements. Happy training! 🚀
